src/interface.jl

```diff
@@ -67,6 +67,12 @@ function update(tree, model, grad, higher...)
   update!(t′, x′, grad, higher...)
 end
 
+function update!(::AbstractRule, model, grad, higher...)
+  error("""update! must be called with an optimiser state, not a rule.
+  Call `opt_state = setup(rule, model)` first, then `update!(opt_state, model, grad)`.
+  """)
```

Comment:
```suggestion
  throw(ArgumentError("""update! must be called with an optimiser state tree, not a rule.
  Call `opt_state = setup(rule, model)` first, then `update!(opt_state, model, grad)`.
  """))
```
To better match language used elsewhere in the docs.

Reply:
The test needs to change to match, I think.